lib/FusionInventory/Agent/Task/Inventory/Linux/Storages.pm

```diff
@@ -13,6 +13,8 @@ use FusionInventory::Agent::Tools::Generic;
 use FusionInventory::Agent::Tools::Linux;
 use FusionInventory::Agent::Tools::Unix;
 
+my $__is_valid_hdparm;
```

Comment:
`memoize` should've worked (i think), but it didn't

Reply:
memoize returns the same result for a given arguments list. I see in _correctHdparmAvailable() you're checking $params{file}. I think this is for testing as we see it in another PR. Then I can confirm memoize won't work if $params{file} changes between calls. And even it won't surely not work during tests but it should definitively on system.

Indeed you moved the _correctHdparmAvailable() call into _getHdparmInfo() and re-use %params but here %params won't be the same on each call as you set device arg during  call. To make memoize work, just don't use %params when calling _correctHdparmAvailable(), but only the really consumed arguments and these ones should not change.

Reply:
memoize didn't work on a real system. I'll try again to make it work.